src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useEffect, useState, Dispatch, SetStateAction } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type TodoModalProps = {
+  todo: Todo;
+  setSelectedTodo: Dispatch<SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  todo,
+  setSelectedTodo,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+  // console.log({ selectedTodoId });
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      setLoading(true);
+      try {
+        const userFromServer = await getUser(todo.userId);
+
+        setUser(userFromServer);
+      } finally {
+        setLoading(false);
+      }
```

Comment:
Consider adding a catch block to handle errors that might occur during the fetch operation. This will help in managing any issues that arise when fetching user data from the server.